cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -689,7 +695,68 @@ Result<std::unique_ptr<substrait::ReadRel>> ScanRelationConverter(
     read_rel_lfs->mutable_items()->AddAllocated(read_rel_lfs_ffs.release());
   }
   read_rel->set_allocated_local_files(read_rel_lfs.release());
-  return std::move(read_rel);
+
+  return Status::OK();
+}
+
+Status SetReadRelationNamedTable(const dataset::InMemoryDataset* dataset,
```

Comment:
```suggestion
Status SetReadRelationNamedTable(const dataset::InMemoryDataset& dataset,
```
The style guide prefers const reference over const pointer:

> Non-optional input parameters should usually be values or const references